src/org/zaproxy/zap/extension/websocket/WebSocketProxy.java

```diff
@@ -668,7 +677,31 @@ public void setForwardOnly(boolean shouldBeForwardOnly) {
 			notifyStateObservers(State.EXCLUDED);
 		}
 	}
-	
+
+	/**
+	 * Sends an ougoing Pong response.
+	 * @param webSocketMessage Ping message was get
+	 */
+	private void sendPongResponse(WebSocketMessage webSocketMessage){
+		if( webSocketMessage instanceof WebSocketProxyV13.WebSocketMessageV13){
+			WebSocketProxyV13.WebSocketMessageV13 webSocketMessageV13 = (WebSocketProxyV13.WebSocketMessageV13) webSocketMessage;
+
+			WebSocketMessageDTO webSocketMessageDTO = webSocketMessageV13.getDTO();
+			webSocketMessageDTO.readableOpcode = WebSocketMessage.opcode2string(WebSocketMessage.OPCODE_PONG);
+			webSocketMessageDTO.opcode = WebSocketMessage.OPCODE_PONG;
+			webSocketMessageDTO.isOutgoing = true;
+			webSocketMessageDTO.id++;
+			try {
+				sendAndNotify(webSocketMessageDTO, Initiator.MANUAL_REQUEST);
```

Comment:
Just noticed that the pongs are not actually being sent (even if shown in the UI), they are not being sent because they are not "changed" :/ (`WebSocketMessageV13.hasChanged` is false after creating the message then in `forward` it does not send anything).